container/container.go

```diff
@@ -58,136 +83,169 @@ func New(opts ...Option) *Container {
 	return cnr
 }
 
-// ToV2 returns the v2 Container message.
-//
-// Nil Container converts to nil.
-func (c *Container) ToV2() *container.Container {
+// CalculateID calculates container identifier
+// based on its structure.
+func CalculateID(c *Container) *cid.ID {
 	if c == nil {
 		return nil
 	}
 
-	return &c.v2
-}
-
-// NewVerifiedFromV2 constructs Container from NeoFS API V2 Container message.
-//
-// Does not perform if message meets NeoFS API V2 specification. To do this
-// use NewVerifiedFromV2 constructor.
-func NewContainerFromV2(c *container.Container) *Container {
-	cnr := new(Container)
-
-	if c != nil {
-		cnr.v2 = *c
-	}
-
-	return cnr
-}
+	var v2 container.Container
+	c.WriteToV2(&v2)
 
-// CalculateID calculates container identifier
-// based on its structure.
-func CalculateID(c *Container) *cid.ID {
-	data, err := c.ToV2().StableMarshal(nil)
+	data, err := v2.StableMarshal(nil)
 	if err != nil {
 		panic(err)
 	}
 
-	id := cid.New()
+	var id cid.ID
 	id.SetSHA256(sha256.Sum256(data))
 
-	return id
+	return &id
 }
 
-func (c *Container) Version() *version.Version {
+// Version returns container's version.
+//
+// Zero Container has version.Current version.
+//
+// See also SetVersion.
+func (c Container) Version() *version.Version {
 	return version.NewFromV2(c.v2.GetVersion())
 }
 
+// SetVersion sets container's version.
+// Version must not be nil.
+//
+// See also Version.
 func (c *Container) SetVersion(v *version.Version) {
```

Comment:
Accept value here and everywhere.

Reply:
https://github.com/nspcc-dev/neofs-sdk-go/pull/186#discussion_r835918884